File: code/JieLi_Home_Demo/NewJieliZhiNeng/GlobalViews/AppSettingVC/AboutVC/DebugView/ShareLog/JLShareDetailViewController.m
//
//  JLShareDetailViewController.m
//  JL_OTA
//
//  Created by EzioChan on 2022/10/17.
//  Copyright © 2022 Zhuhia Jieli Technology. All rights reserved.
//

#import "JLShareDetailViewController.h"

@interface JLShareDetailViewController ()

@property(nonatomic,strong)UITextView *textView;

@end

@implementation JLShareDetailViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self initUI];
}

-(void)initUI{
    self.title = [self.path lastPathComponent];
    UIImage *img = [[UIImage imageNamed:@"Theme.bundle/icon_return.png"] imageWithRenderingMode:UIImageRenderingModeAlwaysOriginal];
    UIBarButtonItem *leftBtn = [[UIBarButtonItem alloc] initWithImage:img style:UIBarButtonItemStyleDone target:self action:@selector(backBtnAction)];
    leftBtn.tintColor = [UIColor grayColor];
    [self.navigationItem setLeftBarButtonItem:leftBtn];
    
    UIBarButtonItem *rightBtn = [[UIBarButtonItem alloc] initWithTitle:@"share" style:UIBarButtonItemStyleDone target:self action:@selector(shareLog)];
    [rightBtn setTintColor:[UIColor blueColor]];
    [self.navigationItem setRightBarButtonItem:rightBtn];
    
    _textView = [UITextView new];
    NSData *data = [NSData dataWithContentsOfFile:self.path];
    NSString *str = [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
    _textView.text = str;
    _textView.font = FontMedium(14);
    _textView.backgroundColor = [UIColor whiteColor];
    _textView.textColor = [UIColor colorFromHexString:@"#606060"];
    _textView.editable = NO;
    [self.view addSubview:_textView];
    
    [self.textView mas_makeConstraints:^(MASConstraintMaker *make) {
        if (@available(iOS 11.0, *)) {
            make.top.equalTo(self.view.mas_safeAreaLayoutGuideTop).offset(16);
            make.bottom.equalTo(self.view.mas_safeAreaLayoutGuideBottom).offset(-16);
        } else {
            make.top.equalTo(self.view.mas_top).offset(16);
            make.bottom.equalTo(self.view.mas_bottom).offset(-16);
        }
        make.left.equalTo(self.view.mas_left).offset(10);
        make.right.equalTo(self.view.mas_right).offset(-10);
    }];
}

-(void)backBtnAction{
    [self.navigationController popViewControllerAnimated:true];
}

-(void)shareLog{
    
    NSArray *arrayItems = @[self.textView.text];
    UIActivityViewController *activeVc = [[UIActivityViewController alloc] initWithActivityItems:arrayItems applicationActivities:nil];
    [self presentViewController:activeVc animated:true completion:nil];
    activeVc.completionWithItemsHandler = ^(UIActivityType _Nullable activityType,BOOL completed,NSArray * _Nullable returnedItems, NSError * _Nullable activityError) {
        
    };
}


@end

File: code/JieLi_Home_Demo/NewJieliZhiNeng/GlobalViews/AppSettingVC/AboutVC/DebugView/ShareLog/JLShareDetailViewController.swift
import UIKit
import SnapKit

class JLShareDetailViewController: DebugBasicViewController {
    
    var path: String = ""
    private var textView: UITextView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
    }
    
    override func initUI() {
        super.initUI()
        
        navigationView.title = (path as NSString).lastPathComponent
        navigationView.rightBtn.isHidden = false
        navigationView.rightBtn.setTitle("share", for: .normal)
        navigationView.rightBtn.setTitleColor(.blue, for: .normal)
        navigationView.rightBtn.addTarget(self, action: #selector(shareLog), for: .touchUpInside)
        
        subTable?.isHidden = true
        
        textView = UITextView()
        if let data = FileManager.default.contents(atPath: path),
           let str = String(data: data, encoding: .utf8) {
            textView.text = str
        }
        textView.font = UIFont.systemFont(ofSize: 14, weight: .medium)
        textView.backgroundColor = .white
        textView.textColor = UIColor(fromHexString: "#606060")
        textView.isEditable = false
        view.addSubview(textView)
        
        textView.snp.makeConstraints { make in
            make.top.equalTo(navigationView.snp.bottom).offset(16)
            make.bottom.equalTo(view.safeAreaLayoutGuide.snp.bottom).offset(-16)
            make.left.right.equalToSuperview().inset(10)
        }
    }
    
    @objc private func shareLog() {
        guard let text = textView.text else { return }
        let activityVc = UIActivityViewController(activityItems: [text], applicationActivities: nil)
        present(activityVc, animated: true, completion: nil)
    }
}
